### DISCUSSION TITLE
persistant volume mount

### DISCUSSION THREAD

**Discussion by devops-eda** [hussein-awala/spark-on-k8s#87] [Q&A]
hello hussein,
i am facing an issue with volume mounting . am unable to mount the volume can you please suggest me the correct way to mount the volume . so that I can pick a file from that mounting path while am submitting a python file.
below are the file currently am using .

```python
import sys
from spark_on_k8s.client import SparkOnK8S, ExecutorInstances, PodResources

# Initialize the SparkOnK8S client
spark_client = SparkOnK8S()

# Configure Spark to access MinIO
spark_conf = {
    "spark.kubernetes.container.image.pullPolicy": "Always",
    "spark.kubernetes.volumes.persistentVolumeClaim.sparky-data.options.claimName": "cafebot-pvc",
    "spark.kubernetes.volumes.persistentVolumeClaim.sparky-data.mount.path": "/opt/cafebot/spark-host",
    "spark.kubernetes.volumes.persistentVolumeClaim.sparky-data.mount.subPath": "spark-host"
}

# Submit the Spark application
spark_client.submit_app(
    image="eda2019/spark-submition:poc-1.7",
    app_path="local:///opt/cafebot/spark-host/process_data.py",
    app_arguments=[],
    app_name="pyspark-job",
    namespace="cafebot",
    service_account="spark",
    app_waiter="log",
    ui_reverse_proxy=True,
    driver_resources=PodResources(cpu=1, memory=512, memory_overhead=128),
    executor_resources=PodResources(cpu=1, memory=512, memory_overhead=128),
    executor_instances=ExecutorInstances(initial=2),
    driver_volume_mounts="/opt/cafebot/spark-host",
    executor_volume_mounts="/opt/cafebot/spark-host",
    spark_conf=spark_conf  # Pass the Spark configuration
)
```

**Comment by hussein-awala** [comment]
You don't need to add the configuration explicitly, you can provide the volumes configurations via `volumes` and `driver_volume_mounts ` and/or `executor_volume_mounts` arguments, and the client will take care of configuring the pod and the executors pods.

I will add an example soon, but you can check https://github.com/hussein-awala/spark-on-k8s/blob/43fab67a350005eabbd31bb6d53c4dda96fce47b/tests/test_spark_client.py#L524-L650 to see how this should be done (`driver/executor_volume_mounts` accept a value of type `list[k8s.V1VolumeMount]` and volumes accepts a value of type `list[k8s.V1Volume]`, where `k8s` is `kubernetes.client`).

We can re-create the issue if it doesn't work as explained above.

**Reply by devops-eda** [reply]
ok let me check nd will update you . Thank you so much for your response .

**Comment by devops-eda** [comment]
Hi hussein,
I wiil try but not able to mount the volume.
below is the python submit file.
-----------------------
import sys
from spark_on_k8s.client import SparkOnK8S, ExecutorInstances, PodResources
from kubernetes import client as k8s

# Define the volume for PersistentVolumeClaim
volumes = k8s.V1Volume(
    name="spark-storage",
    persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name="cafebot-pvc")
)

# Define volume mounts for driver and executor
driver_volume_mounts = k8s.V1VolumeMount(
    name="spark-storage",
    mount_path="/opt/cafebot"
)

executor_volume_mounts = k8s.V1VolumeMount(
    name="spark-storage",
    mount_path="/opt/cafebot"
)

# Initialize the SparkOnK8S client
spark_client = SparkOnK8S()

# Submit the Spark application
try:
    spark_client.submit_app(
        image="spark-submition:poc-1.7",
        app_path="local:///opt/cafebot/spark-host/process_data.py",
        app_arguments=[],
        app_name="pyspark-job",
        namespace="cafebot",
        service_account="spark",
        app_waiter="log",
        ui_reverse_proxy=True,
        driver_resources=PodResources(cpu=1, memory=512, memory_overhead=128),
        executor_resources=PodResources(cpu=1, memory=512, memory_overhead=128),
        executor_instances=ExecutorInstances(initial=2),
        driver_volume_mounts=[driver_volume_mounts],
        executor_volume_mounts=[executor_volume_mounts],
        volumes=[volumes]
    )
except Exception as e:
    print(f"Error submitting Spark application: {e}", file=sys.stderr)
------------------------
while I run this file getting this error 
ERROR:
-------------
java.util.NoSuchElementException: persistentVolumeClaim.spark-storage.options.claimName is required for persistentVolumeClaim
---------------
